### ISSUE TITLE
When uploading a zip, azkaban returns `100-continue` when no credentials or session are provided.

### ISSUE THREAD

**Issue by ryanstull** [azkaban/azkaban#2205]
When uploading a zip to azkaban via the HTTP api, if no credentials are provided, it still returns a `100-continue` upon the initial request.  Then the client, like curl, will upload the zip file, and then nothing will happens!  It looks like azkaban is checking for credentials after the file is uploaded instead of before. 

Also, after the file is uploaded and nothing happens, azkaban returns with a `200` response code, even though this should be a failure case.  But that issue is being tracked here https://github.com/azkaban/azkaban/issues/795

In summary: Azkaban should reject the initial request if it has no authentication or session id instead of responding with `100-continue`.
